### PR TITLE
Update Rootstock 2020 05 18 / Add Spellchecking

### PR DESCRIPTION
**Pull request by cgreene**
### Purpose

This updates rootstock so that we can use new features (github actions builds instead of travis + spellcheck via appveyor).

### Issue

N/A

#### Which areas should receive a particularly close look?

Does it build? Is the spellchecking output useful.

#### Is there anything that you want to discuss further?

Is this catching the type of things we want it to catch?

#### Is the pull request ready for review?

Yes!

### Pull review checklist

Unless otherwise noted above, this PR will be considered ready for review when all four items have been checked.

- [x] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#manubot-markdown)]
- [x] All citations follow the [Manubot citation instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#citations)
- [x] All changes or additions to tables follow the [Manubot table instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#tables)
- [x] All figures follow the [Manubot figure instructions](https://github.com/jaclyn-taroni/OpenPBTA-manuscript/blob/master/USAGE.md#figures)

### CODE REVIEW

**Comment by cgreene** [comment]
```
(base) Caseys-MacBook-Pro:Downloads cgreene$ diff spelling-error-locations-1.0.11-25ff39e.txt spelling-error-locations-1.0.12-0d88536.txt 
12a13
> content/03.methods.md:18:nr
42a44
> content/03.methods.md:45:DINe
54a57
> content/03.methods.md:56:bp
58a62
> content/03.methods.md:59:bp
61a66
> content/03.methods.md:64:hg
80a86
> content/03.methods.md:74:CRAMs
91a98
> content/03.methods.md:80:openPBTA
96a104
> content/03.methods.md:86:VCF
97a106,107
> content/03.methods.md:87:CNVKit
> content/03.methods.md:87:VCF
103a114
> content/03.methods.md:95:PBTA
104a116,122
> content/03.methods.md:98:PBTA
> content/03.methods.md:99:bp
> content/03.methods.md:99:bp
> content/03.methods.md:99:PBTA
> content/03.methods.md:102:VCF
> content/03.methods.md:104:VCF
> content/03.methods.md:105:bp
115a134,135
> content/03.methods.md:108:VCF
> content/03.methods.md:110:GENCODE
116a137,140
> content/03.methods.md:111:bp
> content/03.methods.md:114:MAF
> content/03.methods.md:114:VCF
> content/03.methods.md:116:ENSEMBL
118a143
> content/03.methods.md:121:MAF
122a148,151
> content/03.methods.md:123:OpenPBTA
> content/03.methods.md:124:OpenPBTA
> content/03.methods.md:125:OpenPBTA
> content/03.methods.md:129:GENCODE
123a153
> content/03.methods.md:135:CNVkit
127a158
> content/03.methods.md:139:CNVkit
128a160,164
> content/03.methods.md:140:CNVkit
> content/03.methods.md:141:CNVkit
> content/03.methods.md:141:openpbta
> content/03.methods.md:141:OpenPBTA
> content/03.methods.md:141:OpenPBTA
129a166
> content/03.methods.md:145:CNVKit
130a168,169
> content/03.methods.md:150:bp
> content/03.methods.md:151:bp
140a180
> content/03.methods.md:164:ENSEMBL
141a182
> content/03.methods.md:164:GENCODE
144a186
> content/03.methods.md:167:GENCODE
146a189,191
> content/03.methods.md:171:Ensembl
> content/03.methods.md:171:Ensembl
> content/03.methods.md:171:Gencode
148a194,196
> content/03.methods.md:173:Ensembl
> content/03.methods.md:173:Ensembl
> content/03.methods.md:173:PBTA
152a201
> content/03.methods.md:177:Ensembl
154a204,205
> content/03.methods.md:180:Ensembl
> content/03.methods.md:181:Pearson
164a216
> content/03.methods.md:187:PBTA
179a232
> content/03.methods.md:196:OpenPBTA
180a234
> content/03.methods.md:197:pearson
181a236,237
> content/03.methods.md:199:OpenPBTA
> content/03.methods.md:199:OpenPBTA
191a248
> content/03.methods.md:208:CAVATICA
206a264
> content/03.methods.md:223:OpenPBTA
210a269,270
> content/03.methods.md:225:OpenPBTA
> content/03.methods.md:225:OpenPBTA
220a281
> content/03.methods.md:240:PBTA
224a286
> content/03.methods.md:247:GENCODE
231a294,295
> content/03.methods.md:254:GENCODE
> content/03.methods.md:260:pbta
232a297
> content/03.methods.md:261:pbta
241a307
> content/03.methods.md:306:OpenPBTA
242a309
> content/03.methods.md:310:pbta
248a316
> content/03.methods.md:314:OpenPBTA
262a331
> content/03.methods.md:323:pbta
264a334
> content/03.methods.md:324:OpenPBTA
287a358
> content/03.methods.md:349:Cavatica
288a360,366
> content/04.results.md:7:openpbta
> content/04.results.md:7:openpbta
> content/04.results.md:9:OpenPBTA
> content/04.results.md:11:openpbta
> content/04.results.md:11:openpbta
> content/04.results.md:11:OpenPBTA
> content/04.results.md:15:OpenPBTA
289a368
> content/04.results.md:17:OpenPBTA
290a370
> content/04.results.md:22:openpbta
291a372
> content/04.results.md:25:OpenPBTA
294a376
> content/04.results.md:36:OpenPBTA
(base) Caseys-MacBook-Pro:Downloads cgreene$ 
```

Diffs between the two.

**Comment by cgreene** [comment]
Ok - the results are a little bit odd because of how this is implemented due to an underlying bug in the spell checking software that @agitter implemented a workaround for. First, it finds misspelled words. Then it searches for the instances of those words to annotate them. We've made `ignore-case` false, so it picks up things where there is a casing issue (OpenPBTA for example is sometimes a different case). However, it then annotates _all_ instances of OpenPBTA because the second search step is case insensitive.

**Comment by cgreene** [comment]
This appears to be working as intended now:
```
> content/03.methods.md:260:pbta
232a238
> content/03.methods.md:261:pbta
242c248
< content/03.methods.md:308:Subtyping
---
> content/03.methods.md:310:pbta
262a269
> content/03.methods.md:323:pbta
287a295
> content/03.methods.md:349:Cavatica
288a297,300
> content/04.results.md:7:openpbta
> content/04.results.md:7:openpbta
> content/04.results.md:11:openpbta
> content/04.results.md:11:openpbta
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -29,4 +29,15 @@ Unless otherwise noted above, this PR will be considered ready for review when a
 - [ ] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#manubot-markdown)]
 - [ ] All citations follow the [Manubot citation instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#citations)
 - [ ] All changes or additions to tables follow the [Manubot table instructions](https://github.com/AlexsLemonade/OpenPBTA-manuscript/blob/master/USAGE.md#tables)
-- [ ] All figures follow the [Manubot figure instructions](https://github.com/jaclyn-taroni/OpenPBTA-manuscript/blob/master/USAGE.md#figures)
```
Comment (cgreene) [inline .github/PULL_REQUEST_TEMPLATE.md:32]: Review me

File: README.md [hunk 2]
```diff
@@ -4,7 +4,8 @@
 
 [![HTML Manuscript](https://img.shields.io/badge/manuscript-HTML-blue.svg)](https://AlexsLemonade.github.io/OpenPBTA-manuscript/)
 [![PDF Manuscript](https://img.shields.io/badge/manuscript-PDF-blue.svg)](https://AlexsLemonade.github.io/OpenPBTA-manuscript/manuscript.pdf)
-[![Build Status](https://travis-ci.com/AlexsLemonade/OpenPBTA-manuscript.svg?branch=master)](https://travis-ci.com/AlexsLemonade/OpenPBTA-manuscript)
```
Comment (cgreene) [inline README.md:7]: Review these lines

File: build/assets/custom-dictionary.txt [hunk 3]
```diff
@@ -0,0 +1,245 @@
+personal_ws-1.1 en 244 utf-8
```
Comment (cgreene) [inline build/assets/custom-dictionary.txt:1]: Review these lines

**Comment by cgreene** [comment]
This is finally ready for review 😁

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cgreene) [reply .github/PULL_REQUEST_TEMPLATE.md:32]: FWIW the link in the line below for the Manubot spellchecker won't resolve until this hits master. I wasn't sure of a better way to handle that though.

File: README.md [hunk 4]
```diff
@@ -4,7 +4,8 @@
...
+[![GitHub Actions Status](https://github.com/AlexsLemonade/OpenPBTA-manuscript/workflows/Manubot/badge.svg)](https://github.com/AlexsLemonade/OpenPBTA-manuscript/actions)
+[![Travis Build Status](https://travis-ci.com/AlexsLemonade/OpenPBTA-manuscript.svg?branch=master)](https://travis-ci.com/AlexsLemonade/OpenPBTA-manuscript)
```
Comment (jaclyn-taroni) [inline README.md:8]: So I know Travis is currently disabled, will it continue to be disabled and if so, will this status always indicate that the build is unknown?

Comment (cgreene) [reply README.md:8]: Hmm - probably. Let's just get rid of this :)

**Pull request merged**